main/src/ca/uwaterloo/flix/language/errors/CodeHint.scala

```diff
@@ -41,6 +41,50 @@ object CodeHint {
     def severity: Severity = Severity.Info
   }
 
+  /**
+    * A code hint that indicates an experimental feature.
+    *
+    * @param loc the location of the expression.
+    */
+  case class Experimental(loc: SourceLocation) extends CodeHint {
+    def summary: String = s"Experimental."
+
+    def severity: Severity = Severity.Info
+  }
+
+  /**
+    * A code hint that indicates parallelism.
+    *
+    * @param loc the location of the expression.
+    */
+  case class Parallel(loc: SourceLocation) extends CodeHint {
+    def summary: String = s"Parallel."
+
+    def severity: Severity = Severity.Info
+  }
+
+  /**
+    * A code hint that indicates laziness.
+    *
+    * @param loc the location of the expression.
+    */
+  case class Lazy(loc: SourceLocation) extends CodeHint {
+    def summary: String = s"Lazy."
+
+    def severity: Severity = Severity.Info
+  }
+
+  /**
+    * A code hint that indicates unsafeness.
+    *
+    * @param loc the location of the expression.
+    */
+  case class Unsafe(loc: SourceLocation) extends CodeHint {
+    def summary: String = s"Unsafe."
+
+    def severity: Severity = Severity.Error
+  }
+
   /**
     * A code hint that indicates that a purity polymorphic operation is lazy.
     *
```

Comment:
I think we should keep this file sorted, i.e. Lazy should be just before LazyIfParallel etc.